ovh/resource_cloud_project_network_private.go

```diff
@@ -217,20 +217,81 @@ func resourceCloudProjectNetworkPrivateRead(d *schema.ResourceData, meta interfa
 func resourceCloudProjectNetworkPrivateUpdate(d *schema.ResourceData, meta interface{}) error {
 	config := meta.(*Config)
 	serviceName := d.Get("service_name").(string)
+	regions, _ := helpers.StringsFromSchema(d, "regions")
+
 	params := &CloudProjectNetworkPrivateUpdateOpts{
-		Name: d.Get("name").(string),
+		Name:    d.Get("name").(string),
+		Regions: regions,
+	}
+
+	log.Printf("[DEBUG] params %s", params)
+
+	for _, reg := range params.Regions {
+		param := CloudProjectNetworkPrivateUpdateOptsAlone{
+			Region: reg,
+		}
+		log.Printf("[DEBUG] Will update public cloud private network: %s", param)
+		endpoint := fmt.Sprintf("/cloud/project/%s/network/private/%s/region", serviceName, d.Id())
+		err := config.OVHClient.Post(endpoint, param, nil)
+		if err != nil {
+			if strings.Contains(err.Error(), "already activated") {
+				log.Printf("[DEBUG] Region %s already activated", reg)
+			} else {
+				return fmt.Errorf("calling %s with params %s:\n\t %q", endpoint, param, err)
+			}
+		}
+
+		log.Printf("[DEBUG] Waiting for Private Network %s:", reg)
+
+		stateConf := &resource.StateChangeConf{
+			Pending:    []string{"BUILDING"},
+			Target:     []string{"ACTIVE"},
+			Refresh:    waitForCloudProjectNetworkPrivateActive(config.OVHClient, serviceName, d.Id()),
+			Timeout:    10 * time.Minute,
+			Delay:      10 * time.Second,
+			MinTimeout: 3 * time.Second,
+		}
+
+		if _, err := stateConf.WaitForState(); err != nil {
+			return fmt.Errorf("waiting for private network (%s): %s", params, err)
+		}
+		log.Printf("[DEBUG] Created Private Network %s", reg)
 	}
 
-	log.Printf("[DEBUG] Will update public cloud private network: %s", params)
+	log.Printf("[DEBUG] Will read public cloud private network for project: %s, id: %s", serviceName, d.Id())
 
 	endpoint := fmt.Sprintf("/cloud/project/%s/network/private/%s", serviceName, d.Id())
+	r := &CloudProjectNetworkPrivateResponse{}
+	if err := config.OVHClient.Get(endpoint, r); err != nil {
+		return helpers.CheckDeleted(d, err, endpoint)
+	}
 
-	if err := config.OVHClient.Put(endpoint, params, nil); err != nil {
-		return fmt.Errorf("calling %s with params %s:\n\t %q", endpoint, params, err)
+	regions_attributes := make([]string, 0)
```

Comment:
```suggestion
	currentRegions := make([]string, 0)
```

or something like that ?